Lib/test/test_asyncio/test_queues.py

```diff
@@ -522,5 +522,62 @@ class PriorityQueueJoinTests(_QueueJoinTestMixin, unittest.IsolatedAsyncioTestCa
     q_class = asyncio.PriorityQueue
 
 
+class _QueueShutdownTestMixin:
+    q_class = None
+
+    async def test_empty(self):
+        q = self.q_class()
+        q.shutdown()
+        try:
+            await q.put("data")
+            self.fail("Didn't appear to shut-down queue")
+        except asyncio.QueueShutDown:
+            pass
+        try:
+            await q.get()
+            self.fail("Didn't appear to shut-down queue")
+        except asyncio.QueueShutDown:
+            pass
+
+    async def test_nonempty(self):
+        q = self.q_class()
+        q.put_nowait("data")
+        q.shutdown()
+        await q.get()
+        try:
+            await q.get()
+            self.fail("Didn't appear to shut-down queue")
+        except asyncio.QueueShutDown:
+            pass
+
+    async def test_immediate(self):
+        q = self.q_class()
+        q.put_nowait("data")
+        q.shutdown(immediate=True)
+        try:
+            await q.get()
+            self.fail("Didn't appear to shut-down queue")
+        except asyncio.QueueShutDown:
+            pass
+
```

Comment:
1. Here your tests never iterate the two deque `_getters` and `_putters` because they are empty. So I propose to add 2 dedicated tests.
2. Moreover, I add a test to check if  `joined` tasks/coros are released.
3. And if there is a new line in `format`, I add an another test to check the `_QueueState` values.


```suggestion
    async def test_repr_shutdown(self):
        q = self.q_class()
        q.shutdown()
        self.assertIn("shutdown", repr(q))

        q = self.q_class()
        q.shutdown(immediate=True)
        self.assertIn("shutdown-immediate", repr(q))

    async def test_get_shutdown_immediate(self):
        results = []
        maxsize = 2
        delay = 1e-3

        async def get_q(q):
            try:
                msg = await q.get()
                results.append(False)
            except asyncio.QueueShutDown:
                results.append(True)
                return True

        async def shutdown(q, delay, immediate):
            await asyncio.sleep(delay)
            q.shutdown(immediate)
            return True

        q = self.q_class(maxsize)
        t = [asyncio.create_task(get_q(q)) for _ in range(maxsize)]
        t += [asyncio.create_task(shutdown(q, delay, True))]
        res = await asyncio.gather(*t)

        self.assertEqual(results, [True]*maxsize)


    async def test_put_shutdown(self):
        maxsize = 2
        results = []
        delay = 1e-3

        async def put_twice(q, delay, msg):
            await q.put(msg)
            await asyncio.sleep(delay)
            try:
                await q.put(msg+maxsize)
                results.append(False)
            except asyncio.QueueShutDown:
                results.append(True)
                return msg

        async def shutdown(q, delay, immediate):
            await asyncio.sleep(delay)
            q.shutdown(immediate)

        q = self.q_class(maxsize)
        t = [asyncio.create_task(put_twice(q, delay, i+1)) for i in range(maxsize)]
        t += [asyncio.create_task(shutdown(q, delay*2, False))]
        res = await asyncio.gather(*t)

        self.assertEqual(results, [True]*maxsize)


    async def test_put_and_join_shutdown(self):
        maxsize = 2
        results = []
        delay = 1e-3

        async def put_twice(q, delay, msg):
            await q.put(msg)
            await asyncio.sleep(delay)
            try:
                await q.put(msg+maxsize)
                results.append(False)
            except asyncio.QueueShutDown:
                results.append(True)
                return msg

        async def shutdown(q, delay, immediate):
            await asyncio.sleep(delay)
            q.shutdown(immediate)

        async def join(q, delay):
            await asyncio.sleep(delay)
            await q.join()
            results.append(True)
            return True

        q = self.q_class(maxsize)
        t = [asyncio.create_task(put_twice(q, delay, i+1)) for i in range(maxsize)]
        t += [asyncio.create_task(shutdown(q, delay*2, True)),
                asyncio.create_task(join(q, delay))]
        res = await asyncio.gather(*t)

        self.assertEqual(results, [True]*(maxsize+1))
```
What are you thinking about ?